src/app/actions/plaid.ts

```diff
@@ -0,0 +1,71 @@
+'use server';
+
+import {
+  Configuration,
+  PlaidApi,
+  Products,
+  PlaidEnvironments,
+  CountryCode,
+} from 'plaid';
+import { getRoles, verify } from '@/lib/jwt';
+
+const client = new PlaidApi(new Configuration({
+  basePath: PlaidEnvironments.sandbox,
+  baseOptions: {
+    headers: {
+      'PLAID-CLIENT-ID': '6625de1ca0f038001c8a2e04',
+      'PLAID-SECRET': process.env.PLAID_SECRET,
+      'Plaid-Version': '2020-09-14',
+    },
+  },
+}));
+
+/**
+ * https://plaid.com/docs/api/tokens/#linktokencreate
+ */
+export async function createLinkToken({
+  userId,
+  accessToken,
+}: {
+  userId: string,
+  accessToken: string,
+}): Promise<string> {
+  await verify(accessToken);
+  if (!(await getRoles(accessToken)).isBeta) {
```

Comment:
I'm only protecting the link token creation as without a link token, there is no moving forward in the link creation so this is enough.